specification/cognitiveservices/Language.AnalyzeText/main.tsp

```diff
@@ -0,0 +1,108 @@
+import "@typespec/rest";
+import "@typespec/http";
+import "@typespec/versioning";
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "./common.tsp";
+import "./models/common.models.tsp";
+import "./models/custom.text.tsp";
+import "./models/entity.linking.tsp";
+import "./models/entity.recognition.tsp";
+import "./models/healthcare.tsp";
+import "./models/key.phrase.extraction.tsp";
+import "./models/language.detection.tsp";
+import "./models/pii.entity.recognition.tsp";
+import "./models/sentiment.analysis.tsp";
+import "./models/summarization.tsp";
+import "./models/custom.healthcare.tsp";
+
+using TypeSpec.Http;
+using TypeSpec.Rest;
+using TypeSpec.Versioning;
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+// using Autorest;
+
+@useAuth(
+  TypeSpec.Http.ApiKeyAuth<
+    TypeSpec.Http.ApiKeyLocation.header,
+    "Ocp-Apim-Subscription-Key"
+  >
+)
+@doc("The language service API is a suite of natural language processing (NLP) skills built with best-in-class Microsoft machine learning algorithms.  The API can be used to analyze unstructured text for tasks such as sentiment analysis, key phrase extraction, language detection and question answering. Further documentation can be found in <a href=\"https://docs.microsoft.com/azure/cognitive-services/language-service/overview\">https://docs.microsoft.com/azure/cognitive-services/language-service/overview</a>.0")
+@service({
+  title: "Microsoft Cognitive Language Service - Text Analysis",
+})
+@server(
+  "{Endpoint}/language",
+  "Single server endpoint",
+  {
+    @doc("Supported Cognitive Services endpoint (e.g., https://<resource-name>.api.cognitiveservices.azure.com).")
+    Endpoint: string,
+  }
+)
+@versioned(Versions)
+namespace Language.AnalyzeText;
+
+@doc("Analyze Conversations Service Versions")
+enum Versions {
+  @doc("Version 2022-05-01")
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  v2022_05_01: "2022-05-01",
+
+  @doc("Version 2023-04-01")
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  v2023_04_01: "2023-04-01",
+
+  @doc("Version 2023-04-15-preview")
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  v2023_04_15_preview: "2023-04-15-preview",
```

Comment:
TypeSpec guidance is to only model your stable versions in the project  + only the latest preview api version. Please remove this version from the spec. Your swagger that corresponds to this version and is checked in under the data-plane/ directory will remain there for historical purposes.

Reply:
@catalinaperalta, I have a question about this. We have new attributes introduced in some models which are still in preview introduced in "2023_04_15_preview" and will be present in the new preview "2023_11_15_preview" as shown below:

Do we tag them with "Versions.v2023_11_15_preview" instead? The "@added(Versions.v2023_11_15_preview)" annotation would convey incorrect information in that case. What is the recommendation here?

```
  @added(Versions.v2023_04_15_preview)
  @doc("Enumeration of PII categories to be excluded in the response.")
  excludePiiCategories?: Array<PiiCategoriesExclude>;

  @added(Versions.v2023_04_15_preview)
  @doc("Character to be used for redaction. Default in '*'.")
  redactionCharacter?: redactionCharacter = redactionCharacter.asterisk;
}
```

Reply:
Thanks for fixing. Looks much better :)